ssl/test/bssl_shim.cc

```diff
@@ -905,6 +1038,30 @@ static bool DoExchange(bssl::UniquePtr<SSL_SESSION> *out_session,
       });
     } while (RetryAsync(ssl, ret));
 
+    CheckSSLTransfer(config, ssl);
```

Comment:
Three things we might consider:
* what happens if we transfer a transfered SSL_SESSION?
* have we run the TLS protocol a bit and then performed the transfer?
* transferred the session just before required rekeying and seen what happenes?

Reply:
> have we run the TLS protocol a bit and then performed the transfer?

Yes, see the test category 2.
```
Ensure the |SSL_peek/read| API returns correct value when the original SSL has buffered read data. e.g.
```